### PR TITLE
cleanup: Remove cilium_isitio sidecar configuration

### PR DESCRIPTION
**Pull request by sayboras**
The last cilium istio image was unmaintained for a long time, it's worth mentioning that cilium/proxy repo is no longer building cilium_istio image. This commit is to remove istio sidecar related configuration in cilium codebase.

Related: https://github.com/cilium/proxy/pull/448
Related: https://github.com/cilium/cilium/pull/25722

### CODE REVIEW

File: Documentation/cmdref/cilium-agent.md [hunk 1]
```diff
@@ -311,7 +311,6 @@ cilium-agent [flags]
       --route-metric int                                          Overwrite the metric used by cilium when adding routes to its 'cilium_host' device
       --routing-mode string                                       Routing mode ("native" or "tunnel") (default "tunnel")
       --service-no-backend-response string                        Response to traffic for a service without backends (default "reject")
-      --sidecar-istio-proxy-image string                          Regular expression matching compatible Istio sidecar istio-proxy container image names (default "cilium/istio_proxy")
```
Comment (mhofstetter) [inline Documentation/cmdref/cilium-agent.md:316]: @sayboras  drive-by note (knowing that it's still in draft): the property is also configured in the [Cilium Configmap](https://github.com/cilium/cilium/blob/main/install/kubernetes/cilium/templates/cilium-configmap.yaml#L430-L432) and the [Helm Chart values](https://github.com/cilium/cilium/blob/main/install/kubernetes/cilium/values.yaml.tmpl#L1779-L1781).

Comment (sayboras) [reply Documentation/cmdref/cilium-agent.md:316]: ah good points, I actually missed the above, thanks a lot.

**Review by gandro — APPROVED** [review]
Looks good for my codeowners

File: daemon/cmd/daemon_main.go [hunk 2]
```diff
@@ -730,10 +730,6 @@ func InitGlobalFlags(cmd *cobra.Command, vp *viper.Viper) {
 	flags.Bool(option.Restore, true, "Restores state, if possible, from previous daemon")
 	option.BindEnv(vp, option.Restore)
 
-	flags.String(option.SidecarIstioProxyImage, k8s.DefaultSidecarIstioProxyImageRegexp,
```
Comment (mhofstetter) [inline daemon/cmd/daemon_main.go:733]: i think we should mention the removed flags and helm values in the upgrade guide. (in addition to https://github.com/cilium/cilium/pull/25722)

Comment (sayboras) [reply daemon/cmd/daemon_main.go:733]: ah good point :+1: 

Let's see if there is other comments, which requires for code change. Otherwise, I will update docs in separate PR, just to avoid CI  re-run.

**Review by mhofstetter — APPROVED** [review]

File: daemon/cmd/daemon_main.go (hunk 2 above)
Comment (sayboras) [reply daemon/cmd/daemon_main.go:733]: Done as part of rebase to solve conflict.

Comment (mhofstetter) [reply daemon/cmd/daemon_main.go:733]: what about mentioning the removed flag `--sidecar-istio-proxy-image` too? (in case someone is using is using Cilium without the Helm Chart)

**Review by qmonnet — APPROVED** [review]

**Review by michi-covalent — APPROVED** [review]

**Review by nathanjsweet — APPROVED** [review]
